### PR TITLE
lsyncd: build on darwin

### PR DESCRIPTION
**Pull request by nmattia**
This adds some CMake flags that allow building lsyncd on Darwin.

###### Motivation for this change

I'd like to use lsyncd on Darwin.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by basvandijk** [comment]
@GrahamcOfBorg build lsyncd

File: pkgs/applications/networking/sync/lsyncd/default.nix [hunk 1]
```diff
@@ -37,7 +41,6 @@ stdenv.mkDerivation rec {
     homepage = "https://github.com/axkibe/lsyncd";
     description = "A utility that synchronizes local directories with remote targets";
     license = licenses.gpl2;
-    platforms = platforms.linux;
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/sync/lsyncd/default.nix:40]: Can you set this to all?

**Comment by SuperSandro2000** [comment]
Shouldn't ofborg mark a darwin rebuild? Maybe check if you missed something.

**Comment by basvandijk** [comment]
@GrahamcOfBorg build lsyncd

File: pkgs/applications/networking/sync/lsyncd/default.nix (hunk 1 above)
Comment (nmattia) [reply pkgs/applications/networking/sync/lsyncd/default.nix:40]: Done!

**Review by basvandijk — APPROVED** [review]
It seems ofborg is stuck on darwin. Let's see if it gets unstuck this time around. If it doesn't we can merge it since you built it on darwin already anyway.

File: pkgs/applications/networking/sync/lsyncd/default.nix [hunk 2]
```diff
@@ -37,7 +41,7 @@ stdenv.mkDerivation rec {
     homepage = "https://github.com/axkibe/lsyncd";
     description = "A utility that synchronizes local directories with remote targets";
     license = licenses.gpl2;
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/sync/lsyncd/default.nix:43]: gpl2Only or gpl2Plus?

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 109981` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>lsyncd</li>
</ul>
</details>

File: pkgs/applications/networking/sync/lsyncd/default.nix (hunk 2 above)
Comment (nmattia) [reply pkgs/applications/networking/sync/lsyncd/default.nix:43]: Uh, no idea! But I haven't touched anything there

Comment (SuperSandro2000) [reply pkgs/applications/networking/sync/lsyncd/default.nix:43]: I still saw it and if we want to remove the old licenses we need to change them at some point.

Comment (SuperSandro2000) [reply pkgs/applications/networking/sync/lsyncd/default.nix:43]:
```suggestion
    license = licenses.gpl2Plus;
```

Their source code states this https://github.com/axkibe/lsyncd/blob/master/inotify.c#L4.

**Pull request merged**